File: Bridge/BridgePatternShape.py

from abc import ABC, abstractmethod

class Shape(ABC): # Abstraction
    def __init__(self, imp):
        self._imp = imp

    @abstractmethod
    def draw(self):
        pass

class Rectangle(Shape):
    def draw(self):
        print("Drawing Rectangle with color %s "%(self._imp.fill()))

class Circle(Shape):
    def draw(self):
        print("Drawing Circle with color %s "%(self._imp.fill()))

class Color(ABC): # Implementor
    @abstractmethod
    def fill(self):
        pass

class Red(Color):
    def fill(self):
        return "Red"

class Green(Color):
    def fill(self):
        return "Green"

class Blue(Color):
    def fill(self):
        return "Blue"

# Client code.
c1 = Red()
abstraction = Circle(c1)
abstraction.draw()

c1 = Green()
abstraction = Rectangle(c1)
abstraction.draw()

"""
Drawing Circle with color Red 
Drawing Rectangle with color Green 
"""

File: Bridge/BridgePatternShape.java
// Abstraction interface
interface Shape {
    void draw();
}

// Implementor interface
interface Color {
    String fill();
}

// Rectangle class
class Rectangle implements Shape {
    private Color imp;

    public Rectangle(Color imp) {
        this.imp = imp;
    }

    @Override
    public void draw() {
        System.out.println("Drawing Rectangle with color " + imp.fill());
    }
}

// Circle class
class Circle implements Shape {
    private Color imp;

    public Circle(Color imp) {
        this.imp = imp;
    }

    @Override
    public void draw() {
        System.out.println("Drawing Circle with color " + imp.fill());
    }
}

// Red class
class Red implements Color {
    @Override
    public String fill() {
        return "Red";
    }
}

// Green class
class Green implements Color {
    @Override
    public String fill() {
        return "Green";
    }
}

// Blue class
class Blue implements Color {
    @Override
    public String fill() {
        return "Blue";
    }
}

// Client code
public class BridgePatternShape {
    public static void main(String[] args) {
        Color c1 = new Red();
        Shape abstraction = new Circle(c1);
        abstraction.draw();

        Color c2 = new Green();
        Shape abstraction2 = new Rectangle(c2);
        abstraction2.draw();
    }
}
